src/Rings/walk.jl

```diff
@@ -0,0 +1,1724 @@
+				break
+			end
+		end
+	end
+	if count == length(gens(G))
+		return true
+	end
+	return false
+end
+
+function dot(v::Vector{Int}, w::Vector{Int})
+	sum = 0
+	for i ∈ 1:length(v)
+		@inbounds sum += v[i] * w[i]
+	end
+	return sum
+end
+
+function ordering_as_matrix(w::Vector{Int}, ord::Symbol)
+	if length(w) > 2
+		if ord == :lex
+			return [
+				w'
+				ident_matrix(length(w))[1:length(w)-1, :]
+			]
+		end
+		if ord == :deglex
+			return [
+				w'
+				ones(Int, length(w))'
+				ident_matrix(length(w))[1:length(w)-2, :]
+			]
+		end
+		if ord == :degrevlex
+			return [
+				w'
+				ones(Int, length(w))'
+				anti_diagonal_matrix(length(w))[1:length(w)-2, :]
+			]
+		end
+		if ord == :revlex
+			return [
+				w'
+				anti_diagonal_matrix(length(w))[1:length(w)-1, :]
+			]
+		end
+	else
+		error("not implemented")
+	end
+end
+
+function change_weight_vector(w::Vector{Int}, M::Matrix{Int})
+	return [
+		w'
+		M[2:length(w), :]
+	]
+end
+
+function insert_weight_vector(w::Vector{Int}, M::Matrix{Int})
+	return [
+		w'
+		M[1:length(w)-1, :]
+	]
+end
+
+function add_weight_vector(w::Vector{Int}, M::Matrix{Int})
+	return [
+		w'
+		M
+	]
+end
+
+function ordering_as_matrix(ord::Symbol, nvars::Int)
+	if ord == :lex
+		return ident_matrix(nvars)
+	end
+	if ord == :deglex
+		return [
+			ones(Int, nvars)'
+			ident_matrix(nvars)[1:nvars-1, :]
+		]
+	end
+	if ord == :degrevlex
+		return [
+			ones(Int, nvars)'
+			anti_diagonal_matrix(nvars)[1:nvars-1, :]
+		]
+	end
+	if ord == :revlex
+		return [
+			w'
+			anti_diagonal_matrix(length(w))[1:length(w)-1, :]
+		]
+	else
+		error("not implemented")
+	end
+end
+
+function deg(p::MPolyRingElem, n::Int)
+	max = 0
+	for mon in Singular.monomials(p)
+		ev = Singular.exponent_vectors(mon)
+		sum = 0
+		for e in ev
+			for i ∈ 1:n
+				sum += e[i]
+			end
+			if (max < sum)
+				max = sum
+			end
+		end
+	end
+	return max
+end
+
+
+function checkInt32(w::Vector{Int})
+	for i ∈ 1:length(w)
+		if tryparse(Int32, string(w[i])) == nothing
+			return false
+		end
+	end
+	return true
+end
+
+# computes the representation of the matrixorder defined by T.
+function representation_vector(G::Oscar.IdealGens, T::Matrix{Int})
+	n = size(T)[1]
+	M = 0
+	for i ∈ 1:n
+		for j ∈ 1:n
+			temp = T[i, j]
+			if M < temp
+				M = temp
+			end
+		end
+	end
+	d0 = 0
+	for g in Singular.gens(G)
+		temp = deg(g, n)
+		if d0 < temp
+			d0 = temp
+		end
+	end
+	d = M * (2 * d0^2 + (n + 1) * d0)
+	w = d^(n - 1) * T[1, :]
+	for i ∈ 2:n
+		w = w + d^(n - i) * T[i, :]
+	end
+	return w
+end
+
+# checks if Gw is an initialform of an ideal corresponding to a face of the Groebner fan with dimension less than n-1.
+function inSeveralCones(Gw::Vector{T}) where {T <: MPolyRingElem}
+	counter = 0
+	for g in Gw
+		if size(collect(Singular.coefficients(g)))[1] > 2
+			return true
+		end
+		if size(collect(Singular.coefficients(g)))[1] == 2
+			counter = counter + 1
+		end
+	end
+	if counter > 1
+		return true
+	end
+	return false
+end
```

Comment:
File should end in a newline
```suggestion
end
```